### ISSUE TITLE
Train result on cifar10 classification dataset without pretrained weight

### ISSUE THREAD

**Issue by MenghaoGuo** [lucidrains/vit-pytorch#12]
Just share result.

Transformer structure 

net = ViT(
    image_size = 32,
    patch_size = 16,
    num_classes = 10,
    dim = 1024,
    depth = 12,
    heads = 8,
    mlp_dim = 2048
)

Result :
Epoch: 197
 [============================ 391/391 ===========================>]  Step: 90ms | Tot: 44s860ms | Loss: 1.394 | Acc: 49.418% (24709/50000)
 [============================ 100/100 ===========================>]  Step: 22ms | Tot: 2s680ms | Loss: 1.387 | Acc: 51.130% (5113/10000)

Epoch: 198
 [============================ 391/391 ===========================>]  Step: 91ms | Tot: 44s277ms | Loss: 1.398 | Acc: 49.366% (24683/50000)
 [============================ 100/100 ===========================>]  Step: 29ms | Tot: 2s703ms | Loss: 1.349 | Acc: 51.000% (5100/10000)

Epoch: 199
 [============================ 391/391 ===========================>]  Step: 91ms | Tot: 44s584ms | Loss: 1.396 | Acc: 49.508% (24754/50000)
 [============================ 100/100 ===========================>]  Step: 29ms | Tot: 2s735ms | Loss: 1.410 | Acc: 49.110% (4911/10000)

**Comment by MenghaoGuo** [comment]
Pretrained-weight seems necessary.

**Comment by lucidrains** [comment]
Hmm, I think your patch size is too large. Try a patch size of 2 or 1

**Comment by MenghaoGuo** [comment]
Thanks for your suggest !
I set  patch size as 8 and get following result.
 [============================ 100/100 ===========================>]  Step: 82ms | Tot: 8s44ms | Loss: 1.183 | Acc: 58.070% (5807/10000)
I set  patch size as 4 and get following result.
 [============================ 100/100 ===========================>]  Step: 124ms | Tot: 12s361ms | Loss: 1.035 | Acc: 62.860% (6286/10000)
Thanks for your great work !

**Comment by lucidrains** [comment]
@MenghaoGuo Thanks! Do you want to try 2 or 1 as well? If that doesn't work, you can also try the image encoding method used by iGPT instead of this crude patch-based

**Comment by nikky4D** [comment]
> @MenghaoGuo Thanks! Do you want to try 2 or 1 as well? If that doesn't work, you can also try the image encoding method used by iGPT instead of this crude patch-based method

I'm unfamiliar with igpt. what is the encoding method you mean? why is it more advanced/better that the patch-based method

**Comment by lucidrains** [comment]
@nikky4D it's largely the same architecture as in ViT https://openai.com/blog/image-gpt/ they basically cluster all the pixel values into 512 means, which they discretize to and use as unique tokens for the transformer

**Comment by kentaroy47** [comment]
https://github.com/kentaroy47/vision-transformers-cifar10
Yep, I get similar results too when training from scratch (70-80% accuracy).
Interested in trying with pretrained weights, should be able to reproduce the results in the paper..
